app/org/sagebionetworks/bridge/services/ScheduledActivityService.java

```diff
@@ -214,6 +229,29 @@ public void deleteActivitiesForUser(String healthCode) {
             .collect(toImmutableList());
     }
     
+    /**
+     * If the client data is being added or removed, or if it is different, then the activity is being 
+     * updated.
+     */
+    private boolean hasUpdatedClientData(ScheduledActivity schActivity, ScheduledActivity dbActivity) {
+        JsonNode schNode = (schActivity == null || schActivity.getClientData() == null) ? 
+                null : schActivity.getClientData();
+        JsonNode dbNode = (dbActivity == null || dbActivity.getClientData() == null) ? 
+                null : dbActivity.getClientData();
+        
+        return (schNode == null && dbNode != null) ||
+               (schNode != null && dbNode == null) ||
+               (schNode != null && dbNode != null && !schNode.equals(dbNode));
```

Comment:
Can be simplified, though I've added a comment because then it's not as obvious to me what it's testing for.